src/docs/message-handling.dox

```diff
@@ -0,0 +1,106 @@
+/*! @page message_handling Message handling
+
+@section message_handling_event_handler Message handling using the WT_EVENT_HANDLER
+
+Specific error and other message handling can be configured by passing an
+implementation of WT_EVENT_HANDLER to ::wiredtiger_open or
+WT_CONNECTION::open_session.
+
+For example, both informational and error messages might be passed to an
+application-specific logging function that added a timestamp and logged
+the message to a file, and error messages might additionally be output to
+the \c stderr file stream.
+
+Additionally, applications will normally handle \c WT_PANIC as a special
+case. WiredTiger will always call the error handler callback with
+\c WT_PANIC in the case of a fatal error requiring database restart,
+however, WiredTiger cannot guarantee applications will see an application
+thread return \c WT_PANIC from a WiredTiger API call. For this reason, a
+correctly-written WiredTiger application will likely specify at least an
+error handler which will immediately exit or otherwise handle fatal errors.
+Note that no further WiredTiger calls are required after an error handler
+is called with \c WT_PANIC (and further calls will themselves immediately
+fail).
+
+The following is a programmatic example of creating a custom event handler:
+
+@snippet ex_event_handler.c Function event_handler
+
+The following is a programmatic example of configuring the previously defined custom event handler:
+
+@snippet ex_event_handler.c Configure event_handler
+
+@section message_handling_output Output format of the WT_EVENT_HANDLER
+
+The messages generated by the WT_EVENT_HANDLER can either be encoded in a flat string style format
+or JSON format.
+
+@subsection message_handling_output_flat_string Flat string format
+
+By default, all the messages going through the WT_EVENT_HANDLER are generated as flat strings. The
+example below shows a message following the flat string format:
+
+@code
+[1637732238:116292][11175:0x123456789012], test_salvage.test_salvage.test_salvage_api(column-fix), file:test_salvage.a.wt, WT_SESSION.salvage: [WT_VERB_SALVAGE_PROGRESS][INFO]: WT_SESSION.salvage
+@endcode
+
+@subsection message_handling_output_json JSON format
+
+When configuring a WiredTiger connection through ::wiredtiger_open or WT_CONNECTION::reconfigure, it
+is possible to configure the WT_EVENT_HANDLER interface to produce JSON encoded messages via the
```

Comment:
Would be good to reference the official JSON spec. I'm thinking just adding in a hyperlink in brackets after the word `JSON` in this sentence 
e.g. `configure the WT_EVENT_HANDLER interface to produce JSON (<a href="http://www.json.org">JavaScript Object Notation</a>) encoded messages`

Reply:
Good idea! Done in 6cbdf51ddd6ff02a890e76121216fce53fc2b4b2